packages/container/lib/container.js

```diff
@@ -299,18 +321,40 @@ function instantiate(container, fullName) {
 
     validationCache[fullName] = true;
 
+    let obj;
+
     if (typeof factory.extend === 'function') {
       // assume the factory was extendable and is already injected
-      return factory.create();
+      obj = factory.create();
     } else {
       // assume the factory was extendable
       // to create time injections
       // TODO: support new'ing for instantiation and merge injections for pure JS Functions
-      return factory.create(injectionsFor(container, fullName));
+      obj = factory.create(injectionsFor(container, fullName));
+
+      if (isEnabled('ember-container-inject-owner')) {
+        injectDeprecatedContainer(obj, container);
+      }
     }
+
+    return obj;
   }
 }
 
+// TODO - remove when Ember reaches v3.0.0
+function injectDeprecatedContainer(object, container) {
+  Object.defineProperty(object, 'container', {
```

Comment:
we could likely install the container in injections via a symbol, then leave this defProp on CoreObject, that way all ember objects injections don't need to do the redefine. (only those that are not decedents of CoreObject would?